ani-cli

```diff
@@ -179,26 +184,50 @@ decrypt_link() {
 }
 
 # chooses the link for the set quality
-get_video_quality() {
+get_video_link() {
 	dpage_url="$1"
 	video_links=$(decrypt_link "$dpage_url")
+	if printf '%s' "$video_links" | grep -q "mp4"; then 
+		video_url=$(get_video_quality_mp4 "$video_links")
+		idx=1
+	else
+		video_url="$video_links"
+		get_video_quality_m3u8
+	fi
+}	
+
+get_video_quality_mp4() {
 	case $quality in
 		best)
-			video_link=$(printf '%s' "$video_links" | head -n 4 | tail -n 1)
-			;;
+			video_url=$(printf '%s' "$1" | head -n 4 | tail -n 1) ;;
 		worst)
-			video_link=$(printf '%s' "$video_links" | head -n 1)
-			;;
+			video_url=$(printf '%s' "$1" | head -n 1) ;;
 		*)
-			video_link=$(printf '%s' "$video_links" | grep -i "${quality}p" | head -n 1)
-			if [ -z "$video_link" ]; then
+			video_url=$(printf '%s' "$1" | grep -i "${quality}p" | head -n 1)
+			if [ -z "$video_url" ]; then
 				err "Current video quality is not available (defaulting to best quality)"
 				quality=best
-				video_link=$(printf '%s' "$video_links" | head -n 4 | tail -n 1)
+				video_url=$(printf '%s' "$1" | head -n 4 | tail -n 1)
 			fi
 			;;
 	esac
-	printf '%s' "$video_link"
+	printf '%s' "$video_url"
+}
+
+get_video_quality_m3u8() {
+	case $quality in
+		worst|360)
+			idx=2 ;;
+		480)
+			idx=3 ;;
+		720)
+			idx=4 ;;
+		1080|best)
+			idx=5 ;;
+		*)
+			idx=5 ;;
+	esac
+	printf '%s' "$video_url" | grep -qE "gogocdn.*m3u.*" && idx=$((idx-1))
```

Comment:
Do m3u8 links always have five indexes? Or what would happen if we requested an out of range index?
This could be tested with old anime from times where HD/fullHD didn't exist

Reply:
Yup.. one of the anime that comes in my mind is ```girls bravo``` which has only 2 streams.. ffmpeg can tackle this by selecting the highest if the desired index number is not present (Which I already implemented).. but I m still finding the same solution for mpv..